adapters/handlers/grpc/auth_test.go

```diff
@@ -0,0 +1,128 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2023 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/weaviate/weaviate/entities/models"
+	"google.golang.org/grpc/metadata"
+)
+
+func TestAuth(t *testing.T) {
+	tests := []struct {
+		name        string
+		buildCtx    func() context.Context
+		shouldErr   bool
+		expectedOut *models.Principal
+		allowAnon   bool
+	}{
+		{
+			name: "nothing provided, anon allowed",
+			buildCtx: func() context.Context {
+				return context.Background()
+			},
+			allowAnon: true,
+			shouldErr: false,
+		},
+		{
+			name: "nothing provided, anon forbidden",
+			buildCtx: func() context.Context {
+				return context.Background()
+			},
+			allowAnon: false,
+			shouldErr: true,
+		},
+		{
+			name: "with md, but nothing usable, anon allowed",
+			buildCtx: func() context.Context {
+				md := metadata.Pairs("unrelated", "unrelated")
+				return metadata.NewIncomingContext(context.Background(), md)
+			},
+			allowAnon: true,
+			shouldErr: false,
+		},
+		{
+			name: "with md, but nothing usable, anon forbidden",
+			buildCtx: func() context.Context {
+				md := metadata.Pairs("unrelated", "unrelated")
+				return metadata.NewIncomingContext(context.Background(), md)
+			},
+			allowAnon: false,
+			shouldErr: true,
+		},
+		{
+			name: "with md, but nothing usable, anon allowed",
+			buildCtx: func() context.Context {
+				md := metadata.Pairs("authorization", "wrong-format")
+				return metadata.NewIncomingContext(context.Background(), md)
+			},
+			allowAnon: true,
+			shouldErr: false,
+		},
+		{
+			name: "with md, but nothing usable, anon forbidden",
+			buildCtx: func() context.Context {
+				md := metadata.Pairs("authorization", "wrong-format")
+				return metadata.NewIncomingContext(context.Background(), md)
+			},
+			allowAnon: false,
+			shouldErr: true,
+		},
+		{
+			name: "with md, and a token",
+			buildCtx: func() context.Context {
+				md := metadata.Pairs("authorization", "Bearer Foo")
+				return metadata.NewIncomingContext(context.Background(), md)
+			},
+			shouldErr:   false,
+			expectedOut: &models.Principal{Username: "Foo"},
+		},
+		{
+			name: "with a token that makes extraction error",
```

Comment:
why is this an error?

Reply:
It's hard-coded like that in the mock composer: https://github.com/weaviate/weaviate/pull/2912/files/910a4883788160e5213a22edd3503b0b4833e4e3#diff-d237fce98f1f8062643024ad079478464ed0052bec3523fdce739703a142b3f2R112-R113

I wanted to make sure we cover the path where the composer errors, simulating that it would error trying to extract a token with an invalid signature or something comparable.